test/io-tests/db_config.sql

```diff
@@ -0,0 +1,55 @@
+CREATE ROLE db_config_authenticator LOGIN NOINHERIT;
```

Comment:
The db-config tests are now running with a completely separate user and not the "main" `postgrest_test_authenticator`, with which all other IO tests are running. This prevents a lot of interaction between the config options that are set here and the other tests that don't rely on it. This will make some follow-up PRs much easier to work on.